File: code/Java/tinyi.java
import interpreter.command.BlocksCommand;
import lexical.LexicalAnalysis;
import syntatic.SyntaticAnalysis;

public class tinyi {
	public static void main(String args[]) {
		if (args.length != 1) {
			System.out.println("Usage: java tinyi [source file]");
			return;
		}

		try (LexicalAnalysis l = new LexicalAnalysis(args[0])) {
			SyntaticAnalysis s = new SyntaticAnalysis(l);

			BlocksCommand program = s.start();
			program.execute();
		} catch (Exception e) {
			System.err.println("Internal error: " + e.getMessage());
		}
	}

}

File: code/C++/tinyi.cpp
#include <iostream>
#include <cstdlib>

#include "lexical/LexicalAnalysis.h"
#include "syntatic/SyntaticAnalysis.h"
#include "interpreter/command/Command.h"

int main(int argc, char* argv[]) {
	if (argc != 2) {
		std::cout << "Usage: " << argv[0] << " [Tiny program]" << std::endl;
		exit(1);
	}

	try {
		Lexeme lex;
		LexicalAnalysis l(argv[1]);
		SyntaticAnalysis s(l);

		Command* cmd = s.start();
		cmd->execute();

		delete cmd;
	} catch (const std::string& error) {
		std::cerr << "error: " << error << std::endl;
	}

	return 0;
}
